src/libcore/iter/sources.rs

```diff
@@ -31,8 +32,41 @@ impl<A: Clone> Iterator for Repeat<A> {
 
     #[inline]
     fn next(&mut self) -> Option<A> { Some(self.element.clone()) }
+
     #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) { (usize::MAX, None) }
+
+    #[inline]
+    fn nth(&mut self, _: usize) -> Option<A> { self.next() }
```

Comment:
I like this one more than the others here, but it's an observable change if `A::clone` has side-effects.  Is that allowable?

Reply:
Oh, I didn't consider that. It seems to me that `A::clone` is an implementation detail that shouldn't be relied upon, but as it's technically observable at the moment, and the documentation doesn't specify otherwise, it might be worth clarifying this.